### PR TITLE
RFop_Api

### PR DESCRIPTION
**Pull request by Vital-jan**
# returns RFop queryset with API parameters:
        #state_id=
        #state= (contains)
        #state_exclude= (not contains)
        #kved_id=
        #kved= (contains)
        #name= (contains)
        #address= (contains)
        #address_exclude= (not contains)

### CODE REVIEW

File: data_ocean/serializers/rfop_serializers.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 from rest_framework import serializers
 
 class RfopSerializer(serializers.Serializer):
-    state = serializers.CharField(max_length=100)
+    state_id = serializers.CharField()
```
Comment (Iva-khar) [inline data_ocean/serializers/rfop_serializers.py:4]: I think this not necessary

File: data_ocean/serializers/rfop_serializers.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
+    state = serializers.CharField()
+    kved_id = serializers.CharField()
```
Comment (Iva-khar) [inline data_ocean/serializers/rfop_serializers.py:6]: The same

File: data_ocean/views/rfop_views.py [hunk 3]
```diff
@@ -7,7 +7,45 @@
 from data_ocean.views.views import Views
```
Comment (Iva-khar) [inline data_ocean/views/rfop_views.py:8]: The same for render, HttpResponse, Response

File: data_ocean/views/rfop_views.py [hunk 4]
```diff
@@ -7,7 +7,45 @@
...
 
 class RfopView(Views):
+
     serializer_class = RfopSerializer
     queryset = Rfop.objects.all()
```
Comment (Iva-khar) [inline data_ocean/views/rfop_views.py:12]: The same

File: data_ocean/views/rfop_views.py [hunk 5]
```diff
@@ -7,7 +7,45 @@
...
     serializer = RfopSerializer(queryset, many=True)
```
Comment (Iva-khar) [inline data_ocean/views/rfop_views.py:13]: The same

File: data_ocean/serializers/rfop_serializers.py (hunk 1 above)
Comment (Vital-jan) [reply data_ocean/serializers/rfop_serializers.py:4]: max-length? yes. Thet`s copypaste chiters)

File: data_ocean/serializers/rfop_serializers.py (hunk 2 above)
Comment (Vital-jan) [reply data_ocean/serializers/rfop_serializers.py:6]: +

File: data_ocean/views/rfop_views.py (hunk 3 above)
Comment (Vital-jan) [reply data_ocean/views/rfop_views.py:8]: ??

File: data_ocean/views/rfop_views.py (hunk 4 above)
Comment (Vital-jan) [reply data_ocean/views/rfop_views.py:12]: ??

File: data_ocean/views/rfop_views.py (hunk 5 above)
Comment (Vital-jan) [reply data_ocean/views/rfop_views.py:13]: ??

Comment (Vital-jan) [reply data_ocean/views/rfop_views.py:13]: Thanks for reviews)

File: data_ocean/views/rfop_views.py [hunk 6]
```diff
@@ -7,7 +7,45 @@
...
+    def get_queryset (self):
+        # returns queryset with API parameters:
+        #state_id=
+        #state= (contains)
+        #state_exclude= (not contains)
+        #kved_id=
+        #kved= (contains)
+        #name= (contains)
+        #address= (contains)
+        #address_exclude= (not contains)
```
Comment (criticus) [inline data_ocean/views/rfop_views.py:25]: https://djangobook.com/course/comments-and-docstrings/

File: data_ocean/views/rfop_views.py [hunk 7]
```diff
@@ -7,7 +7,45 @@
...
+
+        resp = self.queryset.all()
+
+        if self.get_parameters (''):
+
```
Comment (criticus) [inline data_ocean/views/rfop_views.py:30]: what is this for?

File: data_ocean/views/views.py [hunk 8]
```diff
@@ -4,8 +4,24 @@
...
 
 class Views (GenericAPIView):
+    
+    def get_parameters (self, param, int_value = 0):
+        # returns http request parameter specified <param> or None if <param> not send by http request
+        # returns all http request parameters if <param> not defined
+        # if <int_value> = 1, validate value of <param> and returns 0, if that's not integer
+        if param:
+            try:
+                resp = self.request.query_params[param]
```
Comment (criticus) [inline data_ocean/views/views.py:14]: what happens if we get unexpected parameter?

File: script.py [hunk 9]
```diff
@@ -1,14 +1,16 @@
...
 from data_ocean.services.ruo_service import RuoConverter
 from data_ocean.services.ratu_service import RatuConverter
+from data_ocean.services.rfop_service import RfopConverter
 
 if RatuConverter().download_file() == 0:
     if RatuConverter().unzip_file() == 0:
         RatuConverter().rename_files()
-        # RatuConverter().process()
+
+# RatuConverter().process()
```
Comment (criticus) [inline script.py:9]: I don't understand what you're trying to implement here? What do these comments mean?

File: script.py [hunk 10]
```diff
@@ -1,14 +1,16 @@
...
 
 if RfopConverter().download_file() == 0:
     if RfopConverter().unzip_file() == 0:
         RfopConverter().rename_files()
-        # RfopConverter().process()
-        # RuoConverter().process()
\ No newline at end of file
+
+# RfopConverter().process()
+RuoConverter().process()
```
Comment (criticus) [inline script.py:16]: same here

File: update.cfg [hunk 11]
```diff
@@ -1 +0,0 @@
-{"https://data.gov.ua/dataset/75e57837-128b-49e1-a007-5e7dfa7bf6af/resource/e21a1e57-051c-46ea-9c8e-8f30de7d863d/download/": 3326459, "https://data.gov.ua/dataset/b244f35a-e50a-4a80-b704-032c42ba8142/resource/06bbccbd-e19c-40d5-9e18-447b110c0b4c/download/": 448365062}
```
Comment (criticus) [inline update.cfg:1]: why do you delete update.cfg?

File: data_ocean/views/rfop_views.py (hunk 6 above)
Comment (Vital-jan) [reply data_ocean/views/rfop_views.py]: I mean this is list of parameters

File: data_ocean/views/rfop_views.py (hunk 7 above)
Comment (Vital-jan) [reply data_ocean/views/rfop_views.py:22]: For exit if parameters not transfer

File: data_ocean/views/views.py (hunk 8 above)
Comment (Vital-jan) [reply data_ocean/views/views.py:14]: Nothing to do

File: script.py (hunk 9 above)
Comment (Vital-jan) [reply script.py]: For run or not run update specified db

File: update.cfg (hunk 11 above)
Comment (Vital-jan) [reply update.cfg:1]: update.cfg not deleted. This file create automatically and uses for check updates of source files.

File: data_ocean/serializers/rfop_serializers.py (hunk 1 above)
Comment (Iva-khar) [reply data_ocean/serializers/rfop_serializers.py:5]: state_id

File: data_ocean/views/rfop_views.py (hunk 3 above)
Comment (Iva-khar) [reply data_ocean/views/rfop_views.py:8]: it not use

File: data_ocean/views/views.py (hunk 8 above)
Comment (IegorSoboliev) [reply data_ocean/views/views.py:14]: I think we don`t need this model now

File: data_ocean/serializers/rfop_serializers.py (hunk 1 above)
Comment (IegorSoboliev) [reply data_ocean/serializers/rfop_serializers.py:5]: Maybe, it`s better to code and use the new (detail) Rfop model for API?

File: data_ocean/views/rfop_views.py (hunk 6 above)
Comment (Vital-jan) [reply data_ocean/views/rfop_views.py]: fixed

File: data_ocean/serializers/rfop_serializers.py (hunk 1 above)
Comment (Vital-jan) [reply data_ocean/serializers/rfop_serializers.py:5]: May be.

**Comment by Vital-jan** [comment]
closed

**Pull request closed without merging**